agent/src/beerocks/slave/son_slave_thread.cpp

```diff
@@ -4371,11 +4371,31 @@ bool slave_thread::handle_client_steering_request(Socket *sd, ieee1905_1::CmduMe
         message_com::send_cmdu(ap_manager_socket, cmdu_tx);
         return true;
     } else {
-        //TODO Add handling of steering opportunity
+
+        // Handling of steering opportunity
+
+        // NOTE: the implementation below does not actually take the steering
+        // opportunity and tries to steer. Instead, it just reports ACK
+        // and steering-completed.
+        // Taking no action is a legitimate result of steering opporunity request,
+        // and this is what is done here.
+        // Later in time we may actually implement the opportunity to steer.
+
         LOG(DEBUG) << "Request Mode bit is not set - Steering Opportunity";
+
+        // Send an ACK message to the controller
```

Comment:
This comment, on the other hand, is redundant. When you create a message of type ACK_MESSAGE, it's fairly obvious that it's an ACK message. No need to add a comment for that.

Reply:
During the years I found it very useful to read the comments _separately_ from the code.
So in this else clause, if one reads the first comment, and then the two successive comments - there is a story, the _intent_ is clear and well described. One can tackle the intent without reading a single line of code. The next thing to do is to _verify_ the code follows the comments.
In this way we separated the logic from implementation and helped building correct and maintainable code.

Reply:
I found it very useful to read the comments separately from the code.
So in this else clause, if one reads the first comment, and then the two successive comments - there is a story, the intent is clear and well described. One can tackle the intent without reading a single line of code. The next thing to do is to verify the code follows the comments.
In this way we separated the logic from implementation and helped building correct and maintainable code.

Reply:
Still - it is redundant, and this is how most if not all the code is written, so please remove the comment.